selendroid-standalone/src/main/java/io/selendroid/standalone/android/impl/AbstractDevice.java

```diff
@@ -338,14 +341,23 @@ public void forwardPort(int local, int remote) {
     }
   }
 
-  private void forwardSelendroidPort(int port) {
-    forwardPort(port, port);
+  private void forwardSelendroidPort(boolean useRandomLocalPort, int port) {
+    if (useRandomLocalPort) {
+      // Let adb forward bind to random port locally
+      String portStr = ShellCommand.exec(
+        adbCommand("forward", "tcp:0", "tcp:" + port));
```

Comment:
Can you add this behavior to `forwardPort` and return the localport from that method instead? Then you just need to optionally pass 0 to `forwardPort` as the local port and set the member variable based on the return value.